compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/NodeCloner.java

```diff
@@ -348,7 +348,6 @@ private void cloneBLangLiteral(BLangLiteral source, BLangLiteral clone) {
 
         clone.value = source.value;
         clone.originalValue = source.originalValue;
-        clone.isFiniteContext = source.isFiniteContext;
```

Comment:
Clone or original I believe type checking should follow the same and algorithms should not be dependent on order if we change the order between runs. 

1. Shouldn't we propagate finalized values back to cloned nodes as well?

2. Removing this line does not stop cloning but now we have a default value `false` only for this type that we may be fooled to consider in other places as a cloned value.

Reply:
One can refer to https://github.com/ballerina-platform/ballerina-lang/issues/36172#issuecomment-1129216294 for details on this change.

Reply:
Change we agreed in our offline discussion added with [2826b05](https://github.com/ballerina-platform/ballerina-lang/pull/36177/commits/2826b0548e610dbe618158a296ae01b4ff36f2e0).